bellbot_gui/src/gui.py

```diff
@@ -129,22 +131,39 @@ def display(self, state):
 	    strands_webserver.client_utils.display_relative_page(display_no, 'livescreen.html')
 
 class GUI_User_Evaluation(object):
-    def __init__(self):
-        rospy.Service('/bellbot/gui/feedback_done', std_srvs.srv.Empty, self.handle_feedback_done)
+    def __init__(self, http_root):
+        self.sub = None
+        self.http_root = http_root
+        self.feedback_filename = os.path.join(self.http_root, "user_feedback.csv")
+        # rospy.Service('/bellbot/gui/feedback_done', std_srvs.srv.Empty, self.handle_feedback_done)
 
-    def handle_feedback_done(self, req):
-        print "feedback done"
+    def display(self, state=None):
+        if self.sub is not None:
+            self.sub.unregister()
+        strands_webserver.client_utils.display_relative_page(display_no, 'feedback.html')
+        self.sub = rospy.Subscriber("/bellbot_gui_feedback", String, self.cb_got_feedback)
+
+    def cb_got_feedback(self, data):
+        feedback = data.data
+        feedback = feedback.split(",")
+        for i in range(len(feedback)):
+            feedback[i] = int(feedback[i])
+        time_now_str = datetime.strftime(datetime.now(), '%Y-%m-%d %H:%M:%S')
+        line_lst = [time_now_str] + feedback
+        with open(self.feedback_filename, "a") as f:
+            writer = csv.writer(f, delimiter=",", quotechar='"', quoting=csv.QUOTE_NONNUMERIC)
+            writer.writerow(line_lst)
```

Comment:
Your feedback file is in the http_root which I assume is the `www` dir in your package. Please keep in mind that after installing it to `/opt/ros/indigo/` you won't have write permissions to it. I'm afraid you have to find a different location to store that. Maybe `~/.ros/bellbot_feedback`?!